src/tracing/instrumentation/express.js

```diff
@@ -95,3 +104,51 @@ function annotateHttpRootSpanWithError(err) {
 
   span.data.http.error = tracingUtil.getErrorDetails(err);
 }
+
+function shimHandlerRegistration(original) {
+  return function shimmedHandlerRegistration() {
+    var args = [];
+    for (var i = 0; i < arguments.length; i++) {
+      var arg = arguments[i];
+      if (typeof arg === 'function') {
+        args.push(wrapHandler(arg));
+      } else {
+        args.push(arg);
+      }
+    }
+    return original.apply(this, args);
+  };
+}
+
+function wrapHandler(fn) {
+  if (fn.length < 4) {
+    // DO NOT REMOVE UNUSED PARAMETERS
+    // express.js checks parameter count to decide what kind of handler function
+    // it should invoke.
+    return function(req, res, next) { // eslint-disable-line no-unused-vars
+      annotateHttpRootSpanWithPathTemplate(req);
+      return fn.apply(this, arguments);
+    };
+  }
+
+  // DO NOT REMOVE UNUSED PARAMETERS
+  // express.js checks parameter count to decide what kind of handler function
+  // it should invoke.
+  return function(err, req, res, next) { // eslint-disable-line no-unused-vars
+    annotateHttpRootSpanWithPathTemplate(req);
+    return fn.apply(this, arguments);
+  };
+}
+
+function annotateHttpRootSpanWithPathTemplate(req) {
```

Comment:
I think a better name would be `annotateHttpEntrySpanWithPathTemplate`, as it is not necessarily a root span, right?

Reply:
Great point. Changed!